triedb/pathdb/database.go

```diff
@@ -148,6 +149,26 @@ var Defaults = &Config{
 // ReadOnly is the config in order to open database in read only mode.
 var ReadOnly = &Config{ReadOnly: true}
 
+// nodeToHash computes the hash of the given node based on the tree structure.
+func nodeToHash(blob []byte, isVerkle bool) (common.Hash, error) {
+	// Compute the node hash in merkle tree manner
+	if !isVerkle {
+		if len(blob) == 0 {
+			return types.EmptyRootHash, nil
+		}
+		return crypto.Keccak256Hash(blob), nil
+	}
+	// Compute the node hash in verkle tree manner
+	if len(blob) == 0 {
+		return types.EmptyVerkleHash, nil
+	}
+	n, err := verkle.ParseNode(blob, 0)
+	if err != nil {
+		return common.Hash{}, err
+	}
+	return n.Commit().Bytes(), nil
```

Comment:
I think the "default" code path should be non-verkle. So instead of having a `if !isVerkle` special case, make it `if isVerkle` into the special case
```suggestion
	if isVerkle { // Compute the node hash in verkle tree manner
		if len(blob) == 0 {
			return types.EmptyVerkleHash, nil
		}
		if n, err := verkle.ParseNode(blob, 0); err != nil {
			return common.Hash{}, err
		} else {
			return n.Commit().Bytes(), nil
		}
	}
		
	if len(blob) == 0 {
		return types.EmptyRootHash, nil
	}
	return crypto.Keccak256Hash(blob), nil
```

Reply:
Another way to do this would be to define two hashers: 
```golang
 func mptHash(blob []byte) (common.Hash, error) {
	if len(blob) == 0 {
			return types.EmptyRootHash, nil
		}
		return crypto.Keccak256Hash(blob), nil
}
func verkleHash(blob []byte) (common.Hash, error) {
	if len(blob) == 0 {
		return types.EmptyVerkleHash, nil
	}
	n, err := verkle.ParseNode(blob, 0)
	if err != nil {
		return common.Hash{}, err
	}
	return n.Commit().Bytes(), nil
}
```
And then, in e.g `Enalbe`, you do 

```
stored = hasher(rawdb.ReadAccountTrieNode(db.diskdb, nil))
```
you'd just configure the `hasher` to use, to be one of either `mptHash` or `verkleHash`. So whenever you set `db.isVerkle`, you can also set `db.hasher` to use the correct one. 

Just an idea...